File: Baekjoon/Class3/32.1697 숨바꼭질.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim();

const MAX_POS = 100000

class HideAndSeek {
    #seekTime;
    constructor(seeker, hider) {
        const depths = new Array(MAX_POS + 1).fill(-1);
        let bfsHead = 0;
        const bfs = [seeker]; // 너비 우선 탐색
        depths[seeker] = 0;
        while(true) {
            const start = bfs[bfsHead++];
            if(start === hider) // 찾음
                break;
            const ends = [start - 1, start + 1, start * 2]; // 못 찾음
            for (const end of ends) {
                if(end >= 0 && end <= MAX_POS && depths[end] === -1) {
                    bfs.push(end);
                    depths[end] = depths[start] + 1;
                }
            }
        }
        this.#seekTime = depths[hider];
    }
    
    get seekTime() {
        return this.#seekTime;
    }
}

const [n, k] = input.split(" ").map(Number);
console.log(new HideAndSeek(n, k).seekTime);

File: Baekjoon/Class3/32.1697 숨바꼭질.py
from collections import deque
import sys
input = sys.stdin.readline

MAX_POS = 100000

class HideAndSeek:
    def __init__(self, seeker, hider):
        depths = [-1] * (MAX_POS + 1)
        bfs = deque([seeker]) #너비 우선 탐색
        depths[seeker] = 0
        while True:
            start = bfs.popleft()
            if start == hider: #찾음
                break
            ends = (start - 1, start + 1, start * 2) #못 찾음
            for end in ends:
                if 0 <= end <= MAX_POS and depths[end] == -1:
                    bfs.append(end)
                    depths[end] = depths[start] + 1
        self._seek_time = depths[hider]
    
    @property
    def seek_time(self):
        return self._seek_time

def main():
    n, k = map(int, input().split())
    print(HideAndSeek(n, k).seek_time)

main()